Shared/Samples/Snap geometry edits with utility network rules/SnapGeometryEditsWithUtilityNetworkRulesView.swift

```diff
@@ -0,0 +1,280 @@
+                                canSave = canUndo
+                            }
+                        case .saving:
+                            try await model.save()
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .errorAlert(presentingError: $error)
+        }
+        .overlay(alignment: .top) {
+            VStack(alignment: .trailing, spacing: 0) {
+                Text(instructionText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.ultraThinMaterial, ignoresSafeAreaEdges: .horizontal)
+                
+                if let selectedElement = model.selectedElement {
+                    VStack(alignment: .leading) {
+                        LabeledContent("Asset Group:", value: selectedElement.assetGroup.name)
+                        LabeledContent("Asset Type:", value: selectedElement.assetType.name)
+                    }
+                    .fixedSize()
+                    .padding()
+                    .background(.ultraThinMaterial)
+                    .clipShape(.rect(cornerRadius: 10))
+                    .shadow(radius: 3)
+                    .padding(8)
+                    .transition(.move(edge: .trailing))
+                }
+            }
+            .animation(.default, value: model.selectedElement == nil)
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Button("Cancel", systemImage: "xmark") {
+                    model.geometryEditor.stop()
+                    model.resetSelection()
+                }
+                .disabled(model.selectedElement == nil)
+                
+                Spacer()
+                
+                Button("Snap Sources") {
+                    sourcesListIsPresented = true
+                }
+                .disabled(model.snapSourceSettings.isEmpty)
+                .popover(isPresented: $sourcesListIsPresented) {
+                    SnapSourcesList(settings: model.snapSourceSettings)
+                        .presentationDetents([.fraction(0.5)])
+                        .frame(idealWidth: 320, idealHeight: 390)
+                }
+                
+                Spacer()
+                
+                if state == .editing {
+                    Button("Save", systemImage: "checkmark") {
+                        state = .saving
+                    }
+                    .disabled(!canSave)
+                } else {
+                    Button("Edit", systemImage: "pencil") {
+                        state = .editing
+                    }
+                    .disabled(model.selectedElement == nil)
+                }
+            }
+        }
+    }
+}
+
+// MARK: - Helper Views
+
+private extension SnapGeometryEditsWithUtilityNetworkRulesView {
```

Comment:
Putting these views in a private extension rather than making them private at the top level means that everything is indented by one level. I suggest getting rid of the extension and just making the views private.